utils/fixed_decimal/src/decimal.rs

```diff
@@ -340,6 +340,37 @@ impl FixedDecimal {
         self
     }
 
+    /// Zero-pad the number on the left to a particular magnitude
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use fixed_decimal::FixedDecimal;
+    ///
+    /// let mut dec = FixedDecimal::from(42);
+    /// assert_eq!("42", dec.to_string());
+    ///
+    /// dec.padded_left_to(3);
+    /// assert_eq!("0042", dec.to_string());
+    ///
+    /// dec.padded_left_to(2);
+    /// assert_eq!("042", dec.to_string());
+    ///
+    /// dec.padded_left_to(1);
+    /// assert_eq!("42", dec.to_string());
+    ///
+    /// dec.padded_left_to(0);
+    /// assert_eq!("42", dec.to_string());
+    /// ```
+    pub fn padded_left_to(&mut self, mut magnitude: i16) {
```

Comment:
Suggestion: If the client wants to decouple padding and truncation, then I think we should have the following methods:

- `pad_left(&mut self, magnitude: i16)`
- `padded_left(self, magnitude: i16)`
- `truncate_left(&mut self, magnitude: i16)`
- `truncated_left(self, magnitude: i16)`

and delete `pad[ded]_or_truncate[d]_left`.  If the client wants both padding and truncation, they can call both functions in sequence.